src/logger.rs

```diff
@@ -55,8 +55,18 @@ impl<'boot> log::Log for Logger {
 
     fn log(&self, record: &log::Record) {
         if let Some(mut ptr) = self.writer {
-            let writer = unsafe { ptr.as_mut() };
-            let result = DecoratedLog::write(writer, record.level(), record.args());
+            // Assumption is that 4096 byte is big enough for every possible
```

Comment:
This is the one assumption that I'm still pondering. What if 4096 bytes aren't enough? For example, somebody might want to debug print the UEFI system table or some other, even bigger struct, and they'll hit this memory limit.

The current implementation isn't the simplest, but it tries to avoid imposing an additional restriction on the calling code (to limit each line to 4096 bytes).

Reply:
yes, I agree. At first, I was not aware of that the current implementation doesn't have this limit (at least in my tests I didn't experience a limit). Given that, we should leave it as it is and keep the current implementation.

Reply:
> Using a string as format buffer (which is backed by a vector) would remove lots of the current "not so simple" code.

That would certainly be the case, but I'd sacrifice the simplicity of the implementation to avoid incurring an additional cost (memory allocations) on the end user.